src/eterna/AnnotationManager.ts

```diff
@@ -440,12 +433,14 @@ export default class AnnotationManager {
      * Recomputes the display objects of all annotations and layers
      */
     public updateAnnotationViews(): void {
+        // TODO: Is this necessary?
         this.highlights.value = [];
+        this.rehighlightSelectedItems(this.allLayers);
+        this.rehighlightSelectedItems(this.allAnnotations);
 
-        const updatedLayers = this.generateAnnotationDisplayObjects(this.allLayers);
-        const updatedAnnotations = this.generateAnnotationDisplayObjects(this.allAnnotations);
-        this._annotations = updatedAnnotations;
-        this._layers = updatedLayers;
+        // TODO: Is there a cleaner way to cache, or do we really need to at all?
```

Comment:
We might not need to do this anymore as well

Reply:
As I'm looking at it, it seems like the situations where these are called a lot they shouldn't actually be used at all, so I bet we can get away with not caching it if I resolve that

Reply:
I stand corrected. This is still necessary given the way it was written. If you have a more elegant approach, definitely feel free to implement it.